### PR TITLE
Promote RunAsGroup to GA

### PR DESCRIPTION
**Pull request by krmayankk**
/kind feature

**What this PR does / why we need it**:

promote this feature to GA for 1.20 kubernetes/enhancements#213
API is unchanged

**Which issue(s) this PR fixes**:
Fixes #https://github.com/kubernetes/enhancements/issues/213

**Special notes for your reviewer**:

```release-note
The `RunAsGroup` feature has been promoted to GA in this release.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190131-runas-groupid.md

### CODE REVIEW

File: pkg/features/kube_features.go
Comment (liggitt) [inline pkg/features/kube_features.go:703]: when moving features to GA, we should:
1. set `LockToDefault: true` and add a comment tracking removal in 2 releases
2. remove checks for `.Enabled(features.RunAsGroup)` (since it is locked to true)
3. ensure API docs no longer indicate it is a beta feature (I think this is already done for this field)
4. ensure e2e test coverage exists

File: pkg/features/kube_features.go
Comment (liggitt) [inline pkg/features/kube_features.go:252]: leave the beta line as-is to make it easy to see the history, and add the GA line

**Comment by SergeyKanzhelev** [comment]
/cc @mrunalp 

(Mrunal was interested to help with this in this doc: https://docs.google.com/document/d/1-NNfuFMh246_ujQpYBCldLQ9hmqEbukaa79Xe1WGwdQ/edit#)

File: pkg/security/podsecuritypolicy/factory.go
Comment (SergeyKanzhelev) [inline pkg/security/podsecuritypolicy/factory.go:54]: this is always true now. So you need to keep what's inside the `if` body

File: pkg/security/podsecuritypolicy/provider.go
Comment (SergeyKanzhelev) [inline pkg/security/podsecuritypolicy/provider.go:143]: same

**Comment by mrunalp** [comment]
:+1:

File: pkg/security/podsecuritypolicy/factory.go
Comment (krmayankk) [reply pkg/security/podsecuritypolicy/factory.go:54]: done

File: pkg/security/podsecuritypolicy/provider.go
Comment (krmayankk) [reply pkg/security/podsecuritypolicy/provider.go:143]: done

**Comment by SergeyKanzhelev** [comment]
@krmayankk as part of the GA process, can you please also help with the following:

1. Convert KEP into the new format with the `kep.yaml` file (example here: https://github.com/kubernetes/enhancements/pull/2071)
2. Request release managers to add milestone 1.21 to the KEP issue so it will be tracked by release team
3. Add notes to the description of this PR confirming that the graduation requirements were met (you may decide to accumulate results as a do like this: https://docs.google.com/document/d/17nROj6ayPsUpx09mhLrzOkRLgo-8sn6Vgfyy-gmyuo4/edit or any other alternative way like a list of PRs confirming that the graduation criteria was met).

LMK if you need some help with these. Thank you!

File: pkg/features/kube_features.go
Comment (ehashman) [inline pkg/features/kube_features.go:776]: This should be updated to 1.22 now?

**Review by ehashman — COMMENTED** [review]
generally LGTM

File: pkg/features/kube_features.go
Comment (krmayankk) [reply pkg/features/kube_features.go]: done

**Comment by liggitt** [comment]
is there an e2e test for this that should be promoted to a (Linux-only?) conformance test as part of this?

**Comment by krmayankk** [comment]
> is there an e2e test for this that should be promoted to a (Linux-only?) conformance test as part of this?
i see these two e2e tests which are Linux-only and agree, they shouuld be part of conformance. 
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/node/security_context.go#L89
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/node/security_context.go#L118

How do you make a test part of conformance @liggitt  ?

**Comment by krmayankk** [comment]
> > is there an e2e test for this that should be promoted to a (Linux-only?) conformance test as part of this?
> 
> i see these two e2e tests which are Linux-only and agree, they shouuld be part of conformance.
> https://github.com/kubernetes/kubernetes/blob/master/test/e2e/node/security_context.go#L89
> https://github.com/kubernetes/kubernetes/blob/master/test/e2e/node/security_context.go#L118
> 
> How do you make a test part of conformance @liggitt ?

opened a PR to make. RunAsGroup part of Conformance @liggitt  @tallclair @BenTheElder  FYI

https://github.com/kubernetes/kubernetes/pull/98645

**Comment by krmayankk** [comment]
the kep with prr just merged https://github.com/kubernetes/enhancements/pull/1974 , can you take another look @liggitt @tallclair  and tell me what else is needed. Also i have opened a PR for making these tests part of conformance, which will merge after this gets in

File: pkg/features/kube_features.go
Comment (tallclair) [inline pkg/features/kube_features.go:212]: Can we get this in v1.21?

Comment (SergeyKanzhelev) [reply pkg/features/kube_features.go:212]: the KEP is actually assigned to 1.21: https://github.com/kubernetes/enhancements/issues/213

Comment (krmayankk) [reply pkg/features/kube_features.go:212]: correct, do i just need to fix this item here to say `ga: 1.21`  ? Who owns reviewing this ?

Comment (ehashman) [reply pkg/features/kube_features.go:212]: Yes, it should say 1.21. Any feature approver can review this file.

Comment (krmayankk) [reply pkg/features/kube_features.go]: fixed

File: pkg/security/podsecuritypolicy/factory.go
Comment (ehashman) [inline pkg/security/podsecuritypolicy/factory.go:54]: Shouldn't we keep this until the feature flag is removed in 1.22?

**Review by ehashman — CHANGES_REQUESTED** [review]

File: pkg/security/podsecuritypolicy/factory.go
Comment (SergeyKanzhelev) [reply pkg/security/podsecuritypolicy/factory.go:54]: This is example shows that it is not needed: https://github.com/kubernetes/kubernetes/pull/94140/ Flag stays to make sure kubelet starts with this flag set

Comment (krmayankk) [reply pkg/security/podsecuritypolicy/factory.go:54]: So no action required from me right ?

**Review by ehashman — COMMENTED** [review]
/lgtm

**Pull request merged**
